truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java

```diff
@@ -1059,10 +1060,14 @@ void executionEnded() {
      * @throws IOException
      */
     Object evalInContext(SuspendedEvent ev, String code, FrameInstance frameInstance) throws IOException {
-        if (frameInstance == null) {
-            return ACCESSOR.evalInContext(engine, ev, code, debugContext.haltedNode, debugContext.haltedFrame);
-        } else {
-            return ACCESSOR.evalInContext(engine, ev, code, frameInstance.getCallNode(), frameInstance.getFrame(FrameAccess.MATERIALIZE, true).materialize());
+        try {
+            if (frameInstance == null) {
+                return ACCESSOR.evalInContext(engine, ev, code, debugContext.haltedNode, debugContext.haltedFrame);
+            } else {
+                return ACCESSOR.evalInContext(engine, ev, code, frameInstance.getCallNode(), frameInstance.getFrame(FrameAccess.MATERIALIZE, true).materialize());
+            }
+        } catch (KillException kex) {
+            throw new IOException("Killed");
```

Comment:
Please provide a better message. Something like "Evaluation was killed." Also initialize the cause in the IOException constructor.